ddsketch/mapping.py

```diff
@@ -169,14 +175,12 @@ def _cubic_exp2_approx(self, value):
             - 9 * self.A * self.B * self.C
             - 27 * self.A * self.A * (value - exponent)
         )
-        cardano = np.cbrt(
-            (delta_1 - np.sqrt(delta_1 * delta_1 - 4 * delta_0 * delta_0 * delta_0)) / 2
-        )
+        cardano = _cbrt((delta_1 - math.sqrt(delta_1 * delta_1 - 4 * delta_0 * delta_0 * delta_0)) / 2)
         significand_plus_one = (
             -(self.B + cardano + delta_0 / cardano) / (3 * self.A) + 1
         )
         mantissa = significand_plus_one / 2
-        return np.ldexp(mantissa, exponent + 1)
+        return math.ldexp(mantissa, exponent + 1)
```

Comment:
~~~
In [23]: assert 10.7 * (2 << 2) == math.ldexp(10.7, 3)

In [24]: %timeit 10.7 * (2 << 2)
7.06 ns ± 0.227 ns per loop (mean ± std. dev. of 7 runs, 100000000 loops each)

In [25]: %timeit math.ldexp(10.7, 3)
62.1 ns ± 0.823 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
~~~

Reply:
```suggestion
        return mantissa * (2 << exponent)
```

Reply:
Yup looks like we're seeing the ~50ns function call overhead for all these 👍

Reply:
Hmm this one doesn't work I think because `exponent` can be negative